src/System.EventLog/src/System/Diagnostics/SharedUtils.cs

```diff
@@ -0,0 +1,335 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Permissions;
+using System.Security.Principal;
+using System.Text;
+using System.Threading;
+using Microsoft.Win32;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Diagnostics
+{
+    internal static class SharedUtils
+    {
+
+        internal const int UnknownEnvironment = 0;
+        internal const int W2kEnvironment = 1;
+        internal const int NtEnvironment = 2;
+        internal const int NonNtEnvironment = 3;
+        private static volatile int environment = UnknownEnvironment;
+
+        private static Object s_InternalSyncObject;
+        private static Object InternalSyncObject
+        {
+            get
+            {
+                if (s_InternalSyncObject == null)
+                {
+                    Object o = new Object();
+                    Interlocked.CompareExchange(ref s_InternalSyncObject, o, null);
+                }
+                return s_InternalSyncObject;
+            }
+        }
+
+        internal static Win32Exception CreateSafeWin32Exception()
+        {
+            return CreateSafeWin32Exception(0);
+        }
+
+        internal static Win32Exception CreateSafeWin32Exception(int error)
+        {
+            Win32Exception newException = null;
+            SecurityPermission securityPermission = new SecurityPermission(PermissionState.Unrestricted);
```

Comment:
remove securitypermission entirely. the try/finally below will be removable then also

Reply:
Corrected